docs/usage/model_config.md

```diff
@@ -0,0 +1,106 @@
+Behaviour of pydantic can be controlled via the `Config` class on a model.
+
+Options:
+
+**`title`**
+: title for the generated JSON Schema
+
+**`anystr_strip_whitespace`**
+: strip or not trailing and leading whitespace for str & byte types (default: `False`)
+
+**`min_anystr_length`**
+: min length for str & byte types (default: `0`)
+
+**`max_anystr_length`**
+: max length for str & byte types (default: `2 ** 16`)
+
+**`validate_all`**
+: whether or not to validate field defaults (default: `False`)
+
+**`extra`**
+: whether to ignore, allow or forbid extra attributes in model. Can use either string values of `ignore`,
+  `allow` or `forbid`, or use `Extra` enum (default is `Extra.ignore`)
+  
+**`allow_mutation`**
+: whether or not models are faux-immutable, e.g. __setattr__ fails (default: `True`)
+
+**`use_enum_values`**
+: whether to populate models with the `value` property of enums,
+  rather than the raw enum - useful if you want to serialise `model.dict()` later (default: `False`)
+  
+**`fields`**
+: schema information on each field, this is equivilant to
+  using [the schema](schema.md) class (default: `None`)
+  
+**`validate_assignment`**
+: whether to perform validation on assignment to attributes or not (default: `False`)
+
+**`allow_population_by_alias`**
+: whether or not an aliased field may be populated by its name as given by the model
+  attribute, rather than strictly the alias; please be sure to read the warning below before enabling this (default:
+  `False`)
+
+!!! warning
+    Think twice before enabling `allow_population_by_alias`! Enabling it could cause previously correct code to become
```

Comment:
Any chance we could weaken this warning a little bi? Now seems like a good time.

Reply:
#585 yes but I'll do it as a separate PR, also need to rename it to `allow_population_by_field_name`.

Reply:
#875